File: Unidad 1/Metodos_ordenamiento/module-javascript/Insertionsort.js
function Insertionsort(arreglo) {

    let n = arreglo.length;

    for (let i = 1; i < n; i++) {

        let clave = arreglo[i];
        let j = i - 1;

        while (j >= 0 && arreglo[j] > clave) {

            arreglo[j + 1] = arreglo[j];
            j--;
        }

        arreglo[j + 1] = clave;
    }

    return arreglo;
}
module.exports = Insertionsort;

File: Unidad 1/Metodos_ordenamiento/src/metodo_insertion_sort/Insertionsort.java
package metodo_insertion_sort;

public class Insertionsort {

    public static void insertionSort(int[] arreglo) {

        int n = arreglo.length;

        for (int i = 1; i < n; i++) {

            int clave = arreglo[i];
            int j = i - 1;

            while (j >= 0 && arreglo[j] > clave) {

                arreglo[j + 1] = arreglo[j];
                j--;
            }

            arreglo[j + 1] = clave;
        }
    }


}
